spring-cloud-task-core/src/test/java/org/springframework/cloud/task/repository/dao/JdbcTaskExecutionDaoTests.java

```diff
@@ -166,6 +174,120 @@ public void testFindAllDefaultSort()  {
+		assertEquals("FOO1", lastTaskExecution.getTaskName());
+
+		final LocalDateTime dateTime = LocalDateTime.ofInstant(lastTaskExecution.getStartTime().toInstant(), ZoneOffset.UTC);
+
+		assertEquals(2015, dateTime.getYear());
+		assertEquals(Month.FEBRUARY, dateTime.getMonth());
+		assertEquals(22, dateTime.getDayOfMonth());
+		assertEquals(23, dateTime.getHour());
+		assertEquals(59, dateTime.getMinute());
+		assertEquals(0, dateTime.getSecond());
+	}
+
+	@Test
+	@DirtiesContext
+	public void testFindLatestTaskExecutions2()  {
+		initializeRepositoryNotInOrderWithMultipleTaskExecutions();
+		final List<TaskExecution> latestTaskExecutions = dao.getLatestTaskExecutionsByTaskNames("FOO1", "FOO3", "FOO4");
+		assertTrue("Expected only 1 taskExecution but got " + latestTaskExecutions.size(), latestTaskExecutions.size() == 3);
+
+		final LocalDateTime dateTimeFoo1 = LocalDateTime.ofInstant(latestTaskExecutions.get(0).getStartTime().toInstant(), ZoneOffset.UTC);
+
+		assertEquals(2015, dateTimeFoo1.getYear());
+		assertEquals(Month.FEBRUARY, dateTimeFoo1.getMonth());
+		assertEquals(22, dateTimeFoo1.getDayOfMonth());
+		assertEquals(23, dateTimeFoo1.getHour());
+		assertEquals(59, dateTimeFoo1.getMinute());
+		assertEquals(0, dateTimeFoo1.getSecond());
+
+		final LocalDateTime dateTimeFoo3 = LocalDateTime.ofInstant(latestTaskExecutions.get(1).getStartTime().toInstant(), ZoneOffset.UTC);
+
+		assertEquals(2016, dateTimeFoo3.getYear());
+		assertEquals(Month.AUGUST, dateTimeFoo3.getMonth());
+		assertEquals(20, dateTimeFoo3.getDayOfMonth());
+		assertEquals(14, dateTimeFoo3.getHour());
+		assertEquals(45, dateTimeFoo3.getMinute());
+		assertEquals(0, dateTimeFoo3.getSecond());
+
+		final LocalDateTime dateTimeFoo4 = LocalDateTime.ofInstant(latestTaskExecutions.get(2).getStartTime().toInstant(), ZoneOffset.UTC);
+
+		assertEquals(2015, dateTimeFoo4.getYear());
+		assertEquals(Month.FEBRUARY, dateTimeFoo4.getMonth());
+		assertEquals(20, dateTimeFoo4.getDayOfMonth());
+		assertEquals(14, dateTimeFoo4.getHour());
+		assertEquals(45, dateTimeFoo4.getMinute());
+		assertEquals(0, dateTimeFoo4.getSecond());
+	}
+
+	@Test
+	@DirtiesContext
+	public void testFindLatestTaskExecutions3()  {
+		initializeRepositoryNotInOrderWithMultipleTaskExecutions();
+		final List<TaskExecution> latestTaskExecutions = dao.getLatestTaskExecutionsByTaskNames("FOO5");
+		assertTrue("Expected only 1 taskExecution but got " + latestTaskExecutions.size(), latestTaskExecutions.size() == 1);
+		final LocalDateTime dateTime = LocalDateTime.ofInstant(latestTaskExecutions.get(0).getStartTime().toInstant(), ZoneOffset.UTC);
+
+		assertEquals(2015, dateTime.getYear());
+		assertEquals(Month.FEBRUARY, dateTime.getMonth());
+		assertEquals(22, dateTime.getDayOfMonth());
+		assertEquals(23, dateTime.getHour());
+		assertEquals(59, dateTime.getMinute());
+		assertEquals(0, dateTime.getSecond());
+	}
+
+	@Test
+	@DirtiesContext
+	public void testGetLatestTaskExecutionForTaskName1()  {
```

Comment:
Can we rename this to testGetLatestTaskExecutionMissingTaskName (or something like it)